geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/MessageDispatcher.java

```diff
@@ -193,6 +194,15 @@ public boolean isWaitingForReAuthentication() {
     return waitForReAuthenticationStartTime > 0;
   }
 
+  private boolean subjectUpdated = false;
```

Comment:
need to be volatile? looks like multiple thread could be accessing this.

Reply:
done